File: cases/taint/java/object-separation-positive/ObjectSeparationPositive.java
package dataflowbench.taint;

final class ObjectSeparationPositive {
    static final class Holder {
        int value;
    }

    static int dfb_source() { // DFB-SOURCE: object-separation-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: object-separation-sink

    static void run() {
        Holder tainted = new Holder();
        Holder clean = new Holder();
        tainted.value = dfb_source(); // DFB-WITNESS: object-separation-store
        clean.value = 0;
        dfb_sink(tainted.value);
    }
}

File: cases/taint/go/object-separation-positive/object_separation_positive.go
package dataflowbench

type Holder struct {
	Value int
}

func dfb_source() int { // DFB-SOURCE: object-separation-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: object-separation-sink

func run() {
	tainted := Holder{}
	clean := Holder{}
	tainted.Value = dfb_source() // DFB-WITNESS: object-separation-store
	clean.Value = 0
	dfb_sink(tainted.Value)
}
